tools/spice/spice_examples_partners.py

```diff
@@ -0,0 +1,289 @@
+
+
+def latitude_longitude_to_ecef(longitude, latitude, altitude):
+    """
+    Convert geodetic coordinates to rectangular coordinates.
+
+    Earth-Centered, Earth-Fixed (ECEF) coordinates are a
+    Cartesian coordinate system with an origin at the center of the Earth.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+
+    Returns
+    -------
+    rect_coords : list
+        Rectangular coordinates in kilometers.
+    """
+    latitude_radians = spice.convrt(latitude, "DEGREES", "RADIANS")
+    longitude_radians = spice.convrt(longitude, "DEGREES", "RADIANS")
+
+    # Retrieve Earth's radii from SPICE
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.bodvrd
+    radii = spice.bodvrd("EARTH", "RADII", 3)[1]
+    equatorial_radius = radii[0]  # Equatorial radius in km
+    polar_radius = radii[2]  # Polar radius in km
+    flattening = (equatorial_radius - polar_radius) / equatorial_radius
+
+    # Convert geodetic coordinates to rectangular coordinates
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.georec
+    rect_coords = spice.georec(
+        longitude_radians, latitude_radians, altitude, equatorial_radius, flattening
+    )
+
+    return rect_coords
+
+
+def calculate_azimuth_elevation(longitude, latitude, altitude, et, target="IMAP"):
+    """
+    Calculate azimuth and elevation.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+    et : int
+        Time at which the state of the target relative to the observer
+        is to be computed. `et' is expressed as seconds past J2000 TDB.
+    target : str (Optional)
+        The target body. Default is "IMAP".
+
+    Returns
+    -------
+    azimuth : float
+        Azimuth in degrees.
+    elevation : float
+        Elevation in degrees.
+    range_to_target : float
+        Range to the target in kilometers.
+    """
+    observer_position_ecef = latitude_longitude_to_ecef(longitude, latitude, altitude)
+
+    method = "Ellipsoid"  # Only method supported
+    abcorr = "None"  # Aberration correction
+    azccw = False  # Azimuth measured clockwise from the positive y-axis
+    elplsz = True  # Elevation increases from the XY plane toward +Z
+    obsctr = "EARTH"  # Name of the center of motion
+    obsref = "IAU_EARTH"  # Body-fixed, body-centered reference frame wrt obsctr
+
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.azlcpo
```

Comment:
Reading this documentation prompted the following thought...
I don't know what the precision requirements are for pointing the antenna, but to be exact, you actually want to point at were the spacecraft was when it sent the data which is the current time minus the one-way-light time. So you would have to shift your azimuth elevation pointing table by that amount.
Again, I don't think that matters here but I found it interesting to think about.

Reply:
Good point. I looked into spice.azlcpo and there is the parameter abcorr. If set to "LT+S" it accounts for this. So that I think that should solve your concern.

"LT+S"     Correct for one-way light time and
                             stellar aberration using a Newtonian
                             formulation. This option modifies the
                             state obtained with the "LT" option to
                             account for the observer's velocity
                             relative to the solar system
                             barycenter. The result is the apparent
                             state of the target---the position and
                             velocity of the target as seen by the
                             observer.